test/display_serialization_test.cpp

```diff
@@ -0,0 +1,88 @@
+    "  [ 1.5,  2.0)  1 |***********                                                 |\n"
+    "  [ 2.0,  inf]  0 |                                                            |\n"
+    "                  +------------------------------------------------------------+\n"
+    "\n";
+
+const std::string expected_2 =
+    "\n"
+    "                   +------------------------------------------------------------+\n"
+    "  [-inf, -5.0)  0  |                                                            |\n"
+    "  [-5.0, -4.1)  19 |*********************************************************   |\n"
+    "  [-4.1, -3.3)  16 |************************************************            |\n"
+    "  [-3.3, -2.4)  12 |************************************                        |\n"
+    "  [-2.4, -1.6)  14 |******************************************                  |\n"
+    "  [-1.6, -0.7)  14 |******************************************                  |\n"
+    "  [-0.7,  0.1)  10 |******************************                              |\n"
+    "  [ 0.1,  1.0)  15 |*********************************************               |\n"
+    "  [ 1.0,  inf]  0  |                                                            |\n"
+    "                   +------------------------------------------------------------+\n"
+    "\n";
+
+const std::string expected_3 =
+  "\n"
+  "                     +------------------------------------------------------------+\n"
+  "  [-inf, -3.0)  0    |                                                            |\n"
+  "  [-3.0, -2.4)  819  |***************************************                     |\n"
+  "  [-2.4, -1.8)  1181 |********************************************************    |\n"
+  "  [-1.8, -1.2)  854  |****************************************                    |\n"
+  "  [-1.2, -0.6)  1162 |*******************************************************     |\n"
+  "  [-0.6,  0.0)  884  |******************************************                  |\n"
+  "  [ 0.0,  0.6)  1037 |*************************************************           |\n"
+  "  [ 0.6,  1.2)  1020 |************************************************            |\n"
+  "  [ 1.2,  1.8)  1203 |*********************************************************   |\n"
+  "  [ 1.8,  2.4)  844  |****************************************                    |\n"
+  "  [ 2.4,  3.0)  996  |***********************************************             |\n"
+  "  [ 3.0,  inf]  0    |                                                            |\n"
+  "                     +------------------------------------------------------------+\n"
+  "\n";
+} // namespace
+
+void run_tests(const std::string& filename, const std::string& expected) {
+
+  auto h1 = make_histogram(axis::regular<>()); // create an empty histogram
+  auto h2 = decltype(h1)(); // create a default-constructed second histogram
+
+  BOOST_TEST_NE(h1, h2);
```

Comment:
@HDembinski 
I can do it, but I was using serialization especially to provide at least one test more data. I preferred to avoid defining one big vector of samples and instead load it from xml file.
Should I then define _h3_ analogous to _h4_?